client/src/main/java/io/pravega/client/stream/notifications/notifier/EndOfDataNotifier.java

```diff
@@ -29,6 +29,11 @@ public EndOfDataNotifier(final NotificationSystem notifySystem,
                              final ScheduledExecutorService executor) {
         super(notifySystem, executor, synchronizer);
     }
+    
+    @VisibleForTesting
+    public void pollNow() {
```

Comment:
Why do you need this in the tests? Is this related to the checkpoint guarantees somehow?

Reply:
This allows the tests to avoid depending on timing. Before they were sleeping and simply assuming the background thread would run if they waited some number of seconds. By invoking the method in the test's thread it's much faster, and no longer prone to flakes.